app/services/add_users.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class AddUsers 
+  def initialize(course:, usernames:, current_user:)
+    @course = course
+    @usernames = usernames
+    @current_user = current_user
+  end
+
+  def add_all_at_once
+    usernames_list = @usernames.lines.map(&:strip)
+    results = {}
+    usernames_list.each do |username|
+      results[username] = add_one_by_one(username)
+    end
+    UpdateCourseWorker.schedule_edits(course: @course, editing_user: @current_user)
```

Comment:
This should be left in the controller. The job of the service will just be to add the users to the course; updating the on-wiki course is a separate idea, one that we carry out right afterwards when a user submits the username list through the UI, but which we might not want to do if we are calling this service in another context (like that setup script).

That will also remove the need to include current_user in the initializer.